File: client-java/instrumentation-shared/src/test/java/org/evomaster/client/java/instrumentation/shared/TaintInputNameTest.java
package org.evomaster.client.java.instrumentation.shared;

import org.junit.jupiter.api.Test;

import java.util.Locale;

import static org.junit.jupiter.api.Assertions.*;

public class TaintInputNameTest {


    @Test
    public void testBase(){

        String name = TaintInputName.getTaintName(0);
        assertTrue(TaintInputName.isTaintInput(name));
    }

    @Test
    public void testNegativeId(){
        assertThrows(Exception.class, () -> TaintInputName.getTaintName(-1));
    }

    @Test
    public void testInvalidNames(){
        assertFalse(TaintInputName.isTaintInput("foo"));
        assertFalse(TaintInputName.isTaintInput(""));
        assertFalse(TaintInputName.isTaintInput("evomaster"));
        assertFalse(TaintInputName.isTaintInput("evomaster_input"));
        assertFalse(TaintInputName.isTaintInput("evomaster__input"));
        assertFalse(TaintInputName.isTaintInput("evomaster_a_input"));

        assertTrue(TaintInputName.isTaintInput("_EM_42_XYZ_"));
    }

    @Test
    public void testInvalidNamePatterns(){
        String prefix = "_EM_";
        String postfix = "_XYZ_";

        assertFalse(TaintInputName.isTaintInput("foo"));
        assertFalse(TaintInputName.isTaintInput(""));
        assertFalse(TaintInputName.isTaintInput(prefix));
        assertFalse(TaintInputName.isTaintInput(prefix + postfix));
        assertFalse(TaintInputName.isTaintInput(prefix+"a"+postfix));

        assertTrue(TaintInputName.isTaintInput(prefix+"42"+postfix));
    }


    @Test
    public void testIncludes(){

        String name = TaintInputName.getTaintName(0);
        String text = "some prefix " + name + " some postfix";

        assertFalse(TaintInputName.isTaintInput(text));
        assertTrue(TaintInputName.includesTaintInput(text));
    }

    @Test
    public void testUpperLowerCase(){

        String name = TaintInputName.getTaintName(0);

        assertTrue(TaintInputName.isTaintInput(name));
        assertTrue(TaintInputName.includesTaintInput(name));


        assertTrue(TaintInputName.isTaintInput(name.toLowerCase()));
        assertTrue(TaintInputName.includesTaintInput(name.toLowerCase()));
        assertTrue(TaintInputName.isTaintInput(name.toUpperCase()));
        assertTrue(TaintInputName.includesTaintInput(name.toUpperCase()));
    }
}

File: outdated/outdated-client-dotnet/test/EvoMaster.Instrumentation.Tests/Coverage/MethodReplacement/TaintInputNameTest.cs
using System;
using EvoMaster.Instrumentation_Shared;
using Xunit;

namespace EvoMaster.Instrumentation.Tests.Coverage.MethodReplacement {
    public class TaintInputNameTest {
        
        [Fact]
        public void TestBase(){

            var name = TaintInputName.GetTaintName(0);
            Assert.True(TaintInputName.IsTaintInput(name));
        }

        [Fact]
        public void TestNegativeId(){
            Assert.Throws<ArgumentException>(() =>  TaintInputName.GetTaintName(-1));
        }
        
        [Fact]
        public void TestInvalidNames() {
            Assert.False(TaintInputName.IsTaintInput("foo"));
            Assert.False(TaintInputName.IsTaintInput(""));
            Assert.False(TaintInputName.IsTaintInput("evomaster"));
            Assert.False(TaintInputName.IsTaintInput("evomaster_input"));
            Assert.False(TaintInputName.IsTaintInput("evomaster__input"));
            Assert.False(TaintInputName.IsTaintInput("evomaster_a_input"));
            Assert.True(TaintInputName.IsTaintInput("_EM_42_XYZ_"));
        }
        
        [Fact]
        public void TestInvalidNamePatterns(){
            var prefix = "_EM_";
            var postfix = "_XYZ_";

            Assert.False(TaintInputName.IsTaintInput("foo"));
            Assert.False(TaintInputName.IsTaintInput(""));
            Assert.False(TaintInputName.IsTaintInput(prefix));
            Assert.False(TaintInputName.IsTaintInput(prefix + postfix));
            Assert.False(TaintInputName.IsTaintInput(prefix+"a"+postfix));

            Assert.True(TaintInputName.IsTaintInput(prefix+"42"+postfix));
        }


        [Fact]
        public void TestIncludes(){

            var name = TaintInputName.GetTaintName(0);
            var text = "some prefix " + name + " some postfix";

            Assert.False(TaintInputName.IsTaintInput(text));
            Assert.True(TaintInputName.IncludesTaintInput(text));
        }
        
        [Fact]
        public void TestUpperLowerCase(){

            String name = TaintInputName.GetTaintName(0);

            Assert.True(TaintInputName.IsTaintInput(name));
            Assert.True(TaintInputName.IncludesTaintInput(name));


            Assert.True(TaintInputName.IsTaintInput(name.ToLower()));
            Assert.True(TaintInputName.IncludesTaintInput(name.ToLower()));
            Assert.True(TaintInputName.IsTaintInput(name.ToUpper()));
            Assert.True(TaintInputName.IncludesTaintInput(name.ToUpper()));
        }
    }
}